src/lido/infos.ts

```diff
@@ -1,28 +1,103 @@
+import BN from "bn.js";
+import { IInstanceVault, IVaultInfoWrapper } from "../types";
+import { LIDO_ADDRESS, LIDO_PROGRAM_ID, ST_SOL_MINT_ADDRESS } from "./ids";
+import { LIDO_TOKEN_LAYOUT } from "./layout";
+import * as types from ".";
 
 let infos: IInstanceVault;
 
 infos = class InstanceLido {
-  static asymc getAllVaults(connection: Connection): Promise<IVaultInfo[]> {}
-  // TODO: Add wrapper for VaultInfo
-  static async getAllVaultWrappers(connection: Connection): Promise<IVaultInfoWrapper[]> {
+  static async getAllVaults(connection: Connection): Promise<types.VaultInfo[]> {
+    const memcmpFilter = {
+      memcmp: { bytes: ST_SOL_MINT_ADDRESS.toString(), offset: 0 },
+    };
+    const config = {
+      filters: [{ dataSize: 165 }, memcmpFilter],
+    };
+    const accountInfos = await connection.getProgramAccounts(TOKEN_PROGRAM_ID, config);
+    const vaults: types.VaultInfo[] = accountInfos.map((info) => this.parseVault(info.account.data, info.pubkey));
+
+    return vaults;
+  }
+
+  static async getAllVaultWrappers(connection: Connection): Promise<VaultInfoWrapper[]> {
     return (await this.getAllVaults(connection)).map((vault) => new VaultInfoWrapper(vault));
   }
-  static asymc getVault(connection: Connection, vaultId: PublicKey): Promise<IVaultInfo>;
-  static asymc parseVault(data: Buffer, vaultId: PublicKey): IVaultInfo;
-  static asymc getAllDepositors(connection: Connection, userKey: PublicKey): Promise<IDepositorInfo[]>;
-  static asymc getDepositor(connection: Connection, depositorId: PublicKey, userKey?: PublicKey): Promise<IDepositorInfo>;
-  static asymc getDepositorId(vaultId: PublicKey, userKey: PublicKey, programId?: PublicKey): PublicKey;
-  static asymc getDepositorIdWithBump?(
-    vaultId: PublicKey,
-    userKey: PublicKey,
-    programId?: PublicKey
-  ): { pda: PublicKey; bump: number };
-  static asymc parseDepositor(data: Buffer, depositorId: PublicKey): IDepositorInfo;
-
-  static asymc getWithdrawerId?(vaultId: PublicKey, userKey: PublicKey, programId?: PublicKey): PublicKey;
-  static asymc getWithdrawer?(connection: Connection, withdrawerId: PublicKey, userKey?: PublicKey): Promise<IWithdrawerInfo>;
-  parseWithdrawer?(data: Buffer, withdrawerId: PublicKey): IWithdrawerInfo;
-  getAllWithdrawers?(connection: Connection, userKey: PublicKey): Promise<IWithdrawerInfo[]>;
+
+  static async getVault(connection: Connection, vaultId: PublicKey): Promise<types.VaultInfo> {
+    const vaultAccountInfo = await connection.getAccountInfo(vaultId);
+    if (!vaultAccountInfo) throw Error("Error: Cannot get solido token account");
+
+    const vault: types.VaultInfo = this.parseVault(vaultAccountInfo.data, vaultId);
+
+    return vault;
+  }
+
+  static parseVault(data: Buffer, vaultId: PublicKey): types.VaultInfo {
+    // Decode the Token data using AccountLayout
+    const decodeData = AccountLayout.decode(data);
+    const { mint, amount } = decodeData;
```

Comment:
I think you might misunderstand what `vault` and `depositor` stand for.

`Vault` should store vault relevant data which `LIDO_LAYOUT` is the info need to stored here.
After user deposit their SOL to vault, Lido will mint stSOL to user instead of creating some user specific account, so `Depositor` will be their token account.

Reply:
The talk with you was super helpful, @Dappio-emerson 

Thanks a lot :)